File: src/model-validation/validate.py
import pandas as pd
from sklearn.metrics import mean_squared_error
from pathlib import Path
import yaml
import os

config_path = Path(__file__).resolve().parents[2] / "config" / "config.yaml"
config = yaml.safe_load(config_path.read_text())
config["paths"]["root_path"] = str(config_path.parents[2])
data_mode = os.getenv("DATA_MODE", "real")

if data_mode == "real":
    data_path = config["data"]["raw"]
else:
    data_path = Path(__file__).resolve().parents[2] / "test" / "sample.csv"

print("Running with DATA_MODE =", data_mode)
print("Raw data path:", data_path)

df = pd.read_csv(data_path)
preds = df['x'] * 2
print('MSE:', mean_squared_error(df['y'], preds))

File: src/model-validation/validate.R
library(here)
library(yaml)

# Load YAML config
config <- yaml.load_file(here("config", "config.yaml"))
config$paths$root_path = here()
data_mode <- Sys.getenv("DATA_MODE", unset = "real")

if (data_mode == "real") {
  data_path <- config$data$raw
  model_path = paste0(config$model$path, "_", config$model$iteration[[1]]$run_id, ".rds")
} else {
  data_path <- here("test", "sample.csv")
  model_path <- here("test", "testmodel.rds")
}

cat("Raw data path:", data_path, "\n")
cat("Model name:", model_path, "\n")

df <- read.csv(data_path)
mse <- mean((df$y - df$x*2)^2)
print(paste('MSE:', mse))
